### PR TITLE
Remove s3

### PR DESCRIPTION
**Pull request by xuzhu-591**
add dummy type of tekton log storage, remove s3 dependency

### CODE REVIEW

File: pkg/cluster/tekton/collector/collector_no_storage.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		tekton: tekton,
+	}
+}
+
+func (c *NoStorageCollector) Collect(ctx context.Context, pr *v1beta1.PipelineRun,
+	horizonMetaData *global.HorizonMetaData) (*CollectResult, error) {
+	const op = "NoStorageCollector: collect"
+	defer wlog.Start(ctx, op).StopPrint()
+
+	// only collect pipelineRun result and Start and end time
```
Comment (kilosonc) [inline pkg/cluster/tekton/collector/collector_no_storage.go:31]: what's the meaning?

Comment (kilosonc) [reply pkg/cluster/tekton/collector/collector_no_storage.go:31]: start time and end time?

File: pkg/cluster/tekton/collector/collector_no_storage.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	logutil "github.com/horizoncd/horizon/pkg/util/log"
+	"github.com/horizoncd/horizon/pkg/util/wlog"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+type NoStorageCollector struct {
+	tekton tekton.Interface
+}
+
+func NewNoStorageCollector(tekton tekton.Interface) *NoStorageCollector {
```
Comment (kilosonc) [inline pkg/cluster/tekton/collector/collector_no_storage.go:20]: *NoStorageCollector => Interface?

File: pkg/cluster/tekton/collector/collector_s3.go [hunk 3]
```diff
@@ -137,20 +140,57 @@ func (c *S3Collector) Collect(ctx context.Context, pr *v1beta1.PipelineRun, hori
...
+		if err != nil {
+			return nil, perror.WithMessagef(err, "failed to get pipelineRun log from s3")
+		}
+		return &Log{
+			LogBytes: logBytes,
+		}, nil
+	}
+
+	// else, get logs from k8s directly
+	logCh, errCh, err := c.tekton.GetPipelineRunLogByID(ctx, pr.CIEventID)
```
Comment (kilosonc) [inline pkg/cluster/tekton/collector/collector_s3.go:180]: this logic is not that right, collector should not access k8s directly

File: config.yaml [hunk 4]
```diff
@@ -51,6 +51,7 @@ tektonMapper:
     # if you are running it on mac, the path may be looked like '/Users/xxx/.kube/config'
     kubeconfig: "/Users/xxx/.kube/config"
     s3:
+      enabled: false
```
Comment (kilosonc) [inline config.yaml:54]: prefering
buildlogstroge:
  type: s3 or dummy
  accessKey: ""
  secretKey: ""

**Review by kilosonc — CHANGES_REQUESTED** [review]

File: config.yaml [hunk 5]
```diff
@@ -50,7 +50,8 @@ tektonMapper:
     # if you run horizon on local machine, you need to set this to the absolute path of your kubeconfig
     # if you are running it on mac, the path may be looked like '/Users/xxx/.kube/config'
     kubeconfig: "/Users/xxx/.kube/config"
-    s3:
+    logStorage:
```
Comment (iamyeka) [inline config.yaml:53]: Add some comments? Please point out what kinds of value could be taken here and what's the use case of each.

File: pkg/cluster/tekton/collector/collector.go [hunk 6]
```diff
@@ -44,6 +46,13 @@ type (
 	}
 )
 
+type Log struct {
```
Comment (iamyeka) [inline pkg/cluster/tekton/collector/collector.go:49]: Why use a struct like this?

Comment (iamyeka) [reply pkg/cluster/tekton/collector/collector.go:49]: Looks like another Log has been defined in core/controller/pipelinerun/controller.go

**Review by kilosonc — APPROVED** [review]

File: config.yaml [hunk 7]
```diff
@@ -50,7 +50,11 @@ tektonMapper:
...
+      # the following types of log storage are supported:
+      #   s3: default minio is used, and you can also configure your own s3 storage
+      #   dummy: a dummy log storage, build logs are not stored and get them directly from k8s
```
Comment (iamyeka) [inline config.yaml:56]: #   s3: Minio is used by default, and you can also specify your own s3 storage.
      #   dummy: A dummy log storage. Building logs are not stored and Horizon gets them directly from k8s.

**Pull request merged**